### PR TITLE
(dsl): Support `constantScore` query

### PR DESCRIPTION
**Pull request by milicns**
Part of #64

### CODE REVIEW

File: docs/overview/queries/elastic_query_constant_score.md [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+val query: ConstantScoreQuery = constantScore(matchPhrase(field = Document.name, value ="test"))
+```
+
+If you want to change the `boost`, you can use `boost` method:
+```scala
+val queryWithBoost: ConstantScoreQuery = constantScore(matchPhrase(field = Document.name, value ="test")).boost(2.2)
+```
+
+You can find more information about `ConstantScore` query [here](https://www.elastic.co/guide/en/elasticsearch/reference/7.17/query-dsl-constant-score-query.html).
+
```
Comment (drmarjanovic) [inline docs/overview/queries/elastic_query_constant_score.md:30]: Unnecessary new line.

**Review by drmarjanovic — CHANGES_REQUESTED** [review]

File: docs/overview/queries/elastic_query_constant_score.md [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+import zio.elasticsearch.query.ConstantScoreQuery
+import zio.elasticsearch.ElasticQuery._
+```
+
+You can create a `ConstantScore` query with arbitrary query(`MatchPhrase` in this example) using the `constantScore` method in the following manner:
+```scala
+val query: ConstantScoreQuery = constantScore(matchPhrase(field = "name", value = "test"))
+```
+
+You can create a [type-safe](https://lambdaworks.github.io/zio-elasticsearch/overview/overview_zio_prelude_schema) `ConstantScore` query with arbitrary query(`MatchPhrase` in this example) using the `constantScore` method in the following manner:
```
Comment (dbulaja98) [inline docs/overview/queries/elastic_query_constant_score.md:19]:
```suggestion
You can create a [type-safe](https://lambdaworks.github.io/zio-elasticsearch/overview/overview_zio_prelude_schema) `ConstantScore` query with arbitrary [type-safe](https://lambdaworks.github.io/zio-elasticsearch/overview/overview_zio_prelude_schema) query(`MatchPhrase` in this example) using the `constantScore` method in the following manner:
```

File: docs/overview/queries/elastic_query_constant_score.md [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+```scala
+val query: ConstantScoreQuery = constantScore(matchPhrase(field = Document.name, value ="test"))
+```
+
+If you want to change the `boost`, you can use `boost` method:
+```scala
+val queryWithBoost: ConstantScoreQuery = constantScore(matchPhrase(field = Document.name, value ="test")).boost(2.2)
```
Comment (dbulaja98) [inline docs/overview/queries/elastic_query_constant_score.md:26]:
```suggestion
val queryWithBoost: ConstantScoreQuery = constantScore(matchPhrase(field = Document.name, value = "test")).boost(2.2)
```

File: modules/integration/src/test/scala/zio/elasticsearch/HttpExecutorSpec.scala [hunk 4]
```diff
@@ -674,6 +674,32 @@ object HttpExecutorSpec extends IntegrationSpec {
...
+                           .refreshTrue
+                       )
+                  query = constantScore(
+                            matchPhrase(
+                              field = TestDocument.stringField,
+                              value = "test"
+                            )
+                          ).boost(2.1)
+                  res <- Executor.execute(ElasticRequest.search(firstSearchIndex, query)).documentAs[TestDocument]
+                } yield assert(res)(Assertion.contains(document))
```
Comment (dbulaja98) [inline modules/integration/src/test/scala/zio/elasticsearch/HttpExecutorSpec.scala:697]: Add if possible: `assert(res)(!Assertion.contains(secondDocumentId)`.

File: modules/library/src/main/scala/zio/elasticsearch/ElasticQuery.scala [hunk 5]
```diff
@@ -25,6 +25,36 @@ import zio.schema.Schema
 
 object ElasticQuery {
 
+  /**
+   * Constructs a type-safe instance of [[zio.elasticsearch.query.ConstantScoreQuery]] with a specified query.
+   * [[zio.elasticsearch.query.ConstantScoreQuery]] wraps a filter query and returns every matching document with a
+   * relevance score equal to the boost parameter value.
+   *
+   * @param query
+   *   query you wish to run
```
Comment (dbulaja98) [inline modules/library/src/main/scala/zio/elasticsearch/ElasticQuery.scala:34]: Maybe something like this:

query to be wrapped inside of constant score query? Or something similar.

File: modules/library/src/main/scala/zio/elasticsearch/ElasticQuery.scala [hunk 6]
```diff
@@ -25,6 +25,36 @@ import zio.schema.Schema
...
+   * @tparam S
+   *   document for which field query is executed. An implicit `Schema` instance must be in scope
```
Comment (dbulaja98) [inline modules/library/src/main/scala/zio/elasticsearch/ElasticQuery.scala:36]: Maybe
```suggestion
   *   document for which field query is specified for. An implicit `Schema` instance must be in scope
```?

File: modules/library/src/main/scala/zio/elasticsearch/ElasticQuery.scala [hunk 7]
```diff
@@ -25,6 +25,36 @@ import zio.schema.Schema
...
+   * @return
+   *   an instance of [[zio.elasticsearch.query.ConstantScoreQuery]] that represents the constant score query with query
+   *   that must satisfy the criteria.
```
Comment (dbulaja98) [inline modules/library/src/main/scala/zio/elasticsearch/ElasticQuery.scala:39]:
```suggestion
   *   an instance of [[zio.elasticsearch.query.ConstantScoreQuery]] that represents the constant score query with query
   *   that must satisfy the criteria to be performed.
```

File: modules/library/src/main/scala/zio/elasticsearch/ElasticQuery.scala [hunk 8]
```diff
@@ -25,6 +25,36 @@ import zio.schema.Schema
...
+   */
+  final def constantScore[S: Schema](query: ElasticQuery[S]): ConstantScoreQuery[S] =
+    ConstantScore[S](query, boost = None)
```
Comment (dbulaja98) [inline modules/library/src/main/scala/zio/elasticsearch/ElasticQuery.scala:42]: Name first parameter.

File: modules/library/src/main/scala/zio/elasticsearch/ElasticQuery.scala [hunk 9]
```diff
@@ -25,6 +25,36 @@ import zio.schema.Schema
...
+   * relevance score equal to the boost parameter value.
+   *
+   * @param query
+   *   query you wish to run
+   * @return
+   *   an instance of [[zio.elasticsearch.query.ConstantScoreQuery]] that represents the constant score query with query
+   *   that must satisfy the criteria.
+   */
+  final def constantScore(query: ElasticQuery[Any]): ConstantScoreQuery[Any] =
+    ConstantScore[Any](query, boost = None)
```
Comment (dbulaja98) [inline modules/library/src/main/scala/zio/elasticsearch/ElasticQuery.scala:56]: Name first parameter.

File: modules/library/src/main/scala/zio/elasticsearch/ElasticQuery.scala [hunk 10]
```diff
@@ -25,6 +25,36 @@ import zio.schema.Schema
...
+  final def constantScore[S: Schema](query: ElasticQuery[S]): ConstantScoreQuery[S] =
+    ConstantScore[S](query, boost = None)
+
+  /**
+   * Constructs an instance of [[zio.elasticsearch.query.ConstantScoreQuery]] with a specified query.
+   * [[zio.elasticsearch.query.ConstantScoreQuery]] wraps a filter query and returns every matching document with a
+   * relevance score equal to the boost parameter value.
+   *
+   * @param query
+   *   query you wish to run
```
Comment (dbulaja98) [inline modules/library/src/main/scala/zio/elasticsearch/ElasticQuery.scala:50]: Same as above.

File: modules/library/src/main/scala/zio/elasticsearch/ElasticQuery.scala [hunk 11]
```diff
@@ -25,6 +25,36 @@ import zio.schema.Schema
...
+   * @return
+   *   an instance of [[zio.elasticsearch.query.ConstantScoreQuery]] that represents the constant score query with query
+   *   that must satisfy the criteria.
```
Comment (dbulaja98) [inline modules/library/src/main/scala/zio/elasticsearch/ElasticQuery.scala:53]: Same as above.

File: modules/library/src/main/scala/zio/elasticsearch/query/Queries.scala [hunk 12]
```diff
@@ -182,6 +182,21 @@ private[elasticsearch] final case class Bool[S](
   }
 }
 
+sealed trait ConstantScoreQuery[S] extends ElasticQuery[S] with HasBoost[ConstantScoreQuery[S]]
+
+private[elasticsearch] final case class ConstantScore[S](query: ElasticQuery[S], boost: Option[Double])
+    extends ConstantScoreQuery[S] { self =>
+  def boost(value: Double): ConstantScoreQuery[S] =
```
Comment (dbulaja98) [inline modules/library/src/main/scala/zio/elasticsearch/query/Queries.scala:189]: Put empty line between.

File: modules/library/src/test/scala/zio/elasticsearch/ElasticQuerySpec.scala [hunk 13]
```diff
@@ -1873,6 +1903,41 @@ object ElasticQuerySpec extends ZIOSpecDefault {
             assert(queryWithAllParams.toJson(fieldPath = None))(equalTo(expectedWithAllParams.toJson))
           }
         ),
+        test("constantScore") {
```
Comment (dbulaja98) [inline modules/library/src/test/scala/zio/elasticsearch/ElasticQuerySpec.scala:1906]: Some parameters are named, some are not.

**Review by dbulaja98 — CHANGES_REQUESTED** [review]

File: modules/library/src/main/scala/zio/elasticsearch/ElasticQuery.scala [hunk 14]
```diff
@@ -25,6 +25,36 @@ import zio.schema.Schema
...
 
+  /**
+   * Constructs a type-safe instance of [[zio.elasticsearch.query.ConstantScoreQuery]] with a specified query.
+   * [[zio.elasticsearch.query.ConstantScoreQuery]] wraps a filter query and returns every matching document with a
+   * relevance score equal to the boost parameter value.
+   *
+   * @param query
+   *   query to be wrapped inside of constant score query
+   * @tparam S
+   *   document for which field query is specified for. An implicit `Schema` instance must be in scope
```
Comment (dbulaja98) [inline modules/library/src/main/scala/zio/elasticsearch/ElasticQuery.scala:36]:
```suggestion
   *   document for which field query is specified for. An implicit `Schema` instance must be provided in the scope
```

**Pull request merged**